src/Common/EventStore/NullEventStorePointerFactory.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\EventStore;
+
+/**
+ * If the EventStore always stays clean, e.g. by using the CleanEventStore listener,
+ * there's no need to set up infrastructure to keep track of the pointer in the stream.
+ */
+final class NullEventStorePointerFactory implements EventStorePointerFactory
```

Comment:
`GapDetection` in `DoctrineEventStore` needs to be configurable too now. On process start, the stream processor would start fetch `0`, because it doesn't keep track here. The next expected id is `1`, then `2` and so on. When the first event in the batch has id `1000`, the `GapDetection` would hammer the database till has verified that `1000` is actually the next event to process. Although it runs smoothly after that, it gets worse after each restart.